### PR TITLE
RxJava3 Migration

### PR DESCRIPTION
**Pull request by Nivador**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -26,9 +26,9 @@ android {
 dependencies {
     implementation project(path: ':rxandroidbleserver')
 
-    implementation 'io.reactivex.rxjava2:rxjava:2.2.17'
-    implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
-    implementation 'com.github.tbruyelle:rxpermissions:0.10.2'
+    implementation 'io.reactivex.rxjava3:rxjava:3.0.0'
+    implementation 'io.reactivex.rxjava3:rxandroid:3.0.0'
+    implementation 'com.github.niklasvd96:rxpermissions:master-SNAPSHOT'
```
Comment (Steppschuh) [inline app/build.gradle]: Avoid using snapshots to allow reproducible builds. Please create a release and use the version number instead.

File: app/src/main/java/com/nexenio/rxandroidbleserverapp/ExampleActivity.java [hunk 2]
```diff
@@ -1,184 +1,186 @@
...
+
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+
+import timber.log.Timber;
+
+import com.tbruyelle.rxpermissions3.RxPermissions;
+
+
+public class ExampleActivity extends AppCompatActivity {
```
Comment (Steppschuh) [inline app/src/main/java/com/nexenio/rxandroidbleserverapp/ExampleActivity.java:29]: Any idea why this is showing as a completely new file?

**Review by Steppschuh — CHANGES_REQUESTED** [review]
Please change the traget branch to `dev`

File: app/build.gradle (hunk 1 above)
Comment (Nivador) [reply app/build.gradle]: fixed

File: app/src/main/java/com/nexenio/rxandroidbleserverapp/ExampleActivity.java (hunk 2 above)
Comment (Nivador) [reply app/src/main/java/com/nexenio/rxandroidbleserverapp/ExampleActivity.java:29]: Unfortunately not. I tried to fix this by starting a new branch from your latest commit on the master branch and applying the changes I made once again on that branch, but that did not fix it

**Review by Steppschuh — APPROVED** [review]

**Comment by Steppschuh** [comment]
Thanks!

**Pull request merged**